apps/meteor/ee/server/apps/communication/rest.ts

```diff
@@ -1124,14 +1128,48 @@ export class AppsRestApi {
 						return API.v1.failure('Invalid status provided, it must be "status" field and a string.');
 					}
 
-					const prl = manager.getOneById(this.urlParams.id);
+					const baseUrl = orchestrator.getMarketplaceUrl();
+					const { id: appId } = this.urlParams;
+					const { status } = this.bodyParams;
 
+					const headers = getDefaultHeaders();
+
+					const prl = manager.getOneById(appId);
 					if (!prl) {
-						return API.v1.notFound(`No App found by the id of: ${this.urlParams.id}`);
+						return API.v1.notFound(`No App found by the id of: ${appId}`);
+					}
+
+					const storedApp = prl.getStorageItem();
+
+					const { version } = prl.getInfo();
+
+					const { isEnterpriseOnly: isStoredAppEnterpriseOnly } = storedApp.marketplaceInfo as IMarketplaceInfo;
+
+					let isAppEnterpriseOnly = isStoredAppEnterpriseOnly ?? false;
```

Comment:
In the case where this is true, do we still need to make the request to the `marketplace-api`? For me it seems that we don't but I may be missing something